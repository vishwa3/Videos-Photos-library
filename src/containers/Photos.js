import React, { useEffect, useRef } from "react";
import { useDispatch, useSelector } from "react-redux";

import { searchMediaAction } from "../actions/mediaActions";

import "../styles/style.css";
import Loader from "./Loader";
import * as types from '../constants/actionTypes';

export function Photos(props) {
  const query = useRef(null);
  const dispatch = useDispatch();

  useEffect(() => {
    console.log("useEffect");
    dispatch(searchMediaAction("dakota", "image"));
    // query.current.focus();
    // dispatch(searchMediaThunk("rain"));
  }, []);

  const images = useSelector(state => state.img.imageArray);
  console.log("type", typeof images, "  ", "len", images.length);
  const error = useSelector(state => state.errorReducer.error);


  const handleSearch = event => {
    event.preventDefault();
    if (query.current.value !== null && query.current.value !== "") {
      dispatch(searchMediaAction(query.current.value, "image"));
      //dispatch(searchMediaThunk(query.current.value));
      query.current.value = "";
    }
  };
  const handleKeyPress = event => {
    if (event.key === 'Enter' && query.current.value !== null && query.current.value !== "") {
      dispatch(searchMediaAction(query.current.value, "image"));
      query.current.value = "";
    }
  }
  return (
    <div className="container-fluid">
      {!error ? (<div>{images.length > 0 ? (
        <div>
          <div className="searchButton">
            <input
              type="text"
              onKeyPress={handleKeyPress}
              style={{ marginRight: 5 }}
              // ref={ref => (this.query = ref)}
              ref={query}
            />
            <input
              type="submit"
              className="btn btn-primary"
              value="Search Images"
              onClick={handleSearch}
            />
          </div>
          <div>
            <h2 className="heading">Images</h2>
            <div className="content">
              <section className="grid">
                {images.map(image => (
                  <div key={image.id} className={`item item-${Math.ceil(
                    image.height / image.width,
                  )}`}>
                    <img className="img-style" src={image.urls.small} alt={image.alt_description} />
                  </div>
                ))}
              </section>
            </div>
          </div>
        </div>) : (<Loader />)}</div>) : (<div> <div className="searchButton">
          <input
            type="text"
            onKeyPress={handleKeyPress}
            style={{ marginRight: 5 }}
            // ref={ref => (this.query = ref)}
            ref={query}
          />
          <input
            type="submit"
            className="btn btn-primary"
            value="Search Images"
            onClick={handleSearch}
          />
        </div><h3>API Error</h3>{error && <div className="error">{JSON.stringify(error)}</div>}</div>)
      }

    </div>
  )

}

export default Photos;