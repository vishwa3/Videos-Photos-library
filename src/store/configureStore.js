import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import rootReducer from '../reducers/index';
import rootSaga from '../sagas'; // TODO: Next step
import thunk from "redux-thunk";

//  Returns the store instance
// It can  also take initialState argument when provided
const configureStore = () => {
  const sagaMiddleware = createSagaMiddleware(); 
  return {
    ...createStore(rootReducer,
      applyMiddleware(sagaMiddleware,thunk)),
    runSaga: sagaMiddleware.run(rootSaga)
  };
};

export default configureStore;