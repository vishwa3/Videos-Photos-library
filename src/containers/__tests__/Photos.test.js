import React from 'react';
import renderer from 'react-test-renderer';
import Photos from '../Photos';
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure,mount } from 'enzyme';
import * as ReactReduxHooks from '../react-redux-hooks'

configure({adapter: new Adapter()});

/* const mockStore = configureMockStore();
const store = mockStore({}); */

describe("Photos", () => {
  let wrapper,useEffect, store;
  
  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(f => f());
  };


  beforeEach(() => {
    store = configureStore()({
      img: {
        imageArray:[]
      },
      vid: {
        videoArray:[]
      },
      errorReducer: {
        error:null
      }
    });
    useEffect = jest.spyOn(React,"useEffect");
    mockUseEffect();
    mockUseEffect();

    jest.spyOn(ReactReduxHooks,"useSelector").mockImplementation(state => store.getState());

    jest.spyOn(ReactReduxHooks,"useDispatch").mockImplementation(()=>store.dispatch);
  
    wrapper = shallow(<Photos store={store} />);
  })

  describe("on start",()=> {
    it("dispatch search Media Action to store", () => {
      const actions = store.getActions();
      console.log("ac",actions);
      console.log("st",store.getState());
      expect(actions).toEqual([{type:'SEARCH_MEDIA_REQUEST',payload:{searchCategory:"image",searchValue:"dakota"}}])
    })
    it("snapshot of Photos component", () => {
      const component = renderer.create(<Photos store={store} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    })
  })
  

 /*  it("search button click should call handleSearch function", () => {
    
    wrapper.find('#search').simulate('click',{ preventDefault() {} });
    expect
  }) */

})



