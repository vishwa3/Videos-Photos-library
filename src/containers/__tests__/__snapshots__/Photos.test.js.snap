// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos on start snapshot of Photos component 1`] = `
<div
  className="container-fluid"
>
  <div>
     
    <div
      className="searchButton"
    >
      <input
        onKeyPress={[Function]}
        style={
          Object {
            "marginRight": 5,
          }
        }
        type="text"
      />
      <input
        className="btn btn-primary"
        onClick={[Function]}
        type="submit"
        value="Search Images"
      />
    </div>
    <h3>
      Error
    </h3>
    <div
      className="error"
    >
      {"img":{"imageArray":[]},"vid":{"videoArray":[]},"errorReducer":{"error":null}}
    </div>
  </div>
</div>
`;
